frontend/src/bundles/common/components/chat-list/chat-item/chat-list-item.tsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
+
+import { Grid, PageAvatar } from '../../components.js';
+import styles from './styles.module.scss';
+
+const MAX_LENGTH = {
+    start: 0,
+    author: 20,
+    message: 30,
+};
+
+type Properties = {
+    userId: string;
+    username: string;
+    lastMessage?: string;
+    lastMessageDate?: string;
+    avatar?: string;
+    itemSelected?: boolean;
+    onClick?: (id: string) => void;
+};
+
+const shrinkMessage = (text: string, length: number): string => {
+    const newText = text.slice(MAX_LENGTH.start, length);
+    return text.length > length ? newText + '...' : newText;
+};
```

Comment:
Are you sure we can't do it using css text-overflow: ellipsis?

Reply:
I know, I tried and didnt get expected result, so I decided write that. but today I found that for proper working ellipsis we should set fixed width. so I will remove that bulky funtion